cupyx/scipy/ndimage/interpolation.py

```diff
@@ -233,13 +193,31 @@ def affine_transform(input, matrix, offset=0.0, output_shape=None, output=None,
     if output_shape is None:
         output_shape = input.shape
 
-    coordinates = cupy.indices(output_shape, dtype=cupy.float64)
-    coordinates = cupy.dot(matrix, coordinates.reshape((input.ndim, -1)))
-    coordinates += cupy.expand_dims(cupy.asarray(offset), -1)
-    ret = _get_output(output, input, output_shape)
-    ret[:] = map_coordinates(input, coordinates, ret.dtype, order, mode, cval,
-                             prefilter).reshape(output_shape)
-    return ret
+    matrix = matrix.astype(float, copy=False)
```

Comment:
Please use `cupy.float64` for the readability.
```suggestion
    matrix = matrix.astype(cupy.float64, copy=False)
```

Reply:
BTW, Is this typecast needed? A float64 array looks to be returned without this typecast.